masonry/src/widget/widget.rs

```diff
@@ -165,20 +167,29 @@ pub trait Widget: AsAny {
 
     // --- Auto-generated implementations ---
 
-    // FIXME
-    #[cfg(FALSE)]
     /// Return which child, if any, has the given `pos` in its layout rect.
     ///
-    /// The child return is a direct child, not eg a grand-child. The position is in
+    /// The child returned is a direct child, not eg a grand-child. The position is in
     /// relative coordinates. (Eg `(0,0)` is the top-left corner of `self`).
     ///
     /// Has a default implementation, that can be overridden to search children more
     /// efficiently.
-    fn get_child_at_pos(&self, pos: Point) -> Option<WidgetRef<'_, dyn Widget>> {
-        // layout_rect() is in parent coordinate space
-        self.children()
-            .into_iter()
-            .find(|child| child.state().layout_rect().contains(pos))
+    ///
+    /// The child widget references in `children` are in the same order as returned by
+    /// [`Self::children_ids`].
+    fn get_child_at_pos<'w>(
+        &self,
+        children: &[WidgetRef<'w, dyn Widget>],
+        pos: Point,
```

Comment:
> Perhaps `Widget::get_child_at_pos` should mirror that and take `pos` in global coordinates.

Definitely. I'm not a fan of the "remove origin from pos each loop cycle" game. It's too brittle, and we've had bugs because of it before. I'd like to remove it anywhere we can.